### ISSUE TITLE
Autofix flag

### ISSUE THREAD

**Issue by mstruebing** [editorconfig-checker/editorconfig-checker.go#14]
introduce an autofix flag which should fix all issues which can be fixed.

**Comment by clarfonthey** [comment]
Was there a particular reason why the existing PR couldn't be merged?

**Comment by mstruebing** [comment]
I think the main reason was time to complete it and maybe some things I had problems with.
It's so long ago that it would might make more sense to create a new PR for this.

**Comment by clarfonthey** [comment]
That's fair; I mostly asked if there was anything specifically wrong with it besides it just being unfinished, because I was considering looking into working on this and was thinking of looking into the original PR as a base for that.

My thought process was that if we at least have a CLI flag and the framework in the code to do autofixes, even if we don't have fixes for all the rules right away, it will be a start.

**Comment by mstruebing** [comment]
Sounds good 👍 and thank you for considering working on that.

**Comment by kamalmemon** [comment]
Is there any update on this? Incorporating an autofix mechanism would greatly enhance the developer experience

**Comment by klaernie** [comment]
Nobody so far has dared to tackle this. Rebasing the changes from [the old addFix branch](https://github.com/editorconfig-checker/editorconfig-checker/compare/main...addFix) branch would be a start, but developing the strategy on how to fix the found issues in each of validations is the real crux. The biggest problem that comes to my mind is autofixing the indentation, since that can cause semantic problems.

I think editors are generally the better place to solve this problem, since typically they have semantic understanding of the code, especially since many nowadays support language servers.

But if you have time on your hand, I'd love to review that PR.

**Comment by corneliusroemer** [comment]
I think any auto fix capabilities would be better than none. Don't worry about the hard cases, start with the simple ones: line ending characters and newline at end. Those should be easy. Can add .kee later.

**Comment by zeitlinger** [comment]
Adding some current context and checking whether a contribution would be welcome.

We are evaluating `editorconfig-checker` as a replacement for a small “miscellaneous files” formatting block currently handled by Spotless. The useful safe fixes for this use case would be:

- trim trailing whitespace;
- normalize line endings;
- ensure a final newline (and avoid accumulating extra trailing blank lines, if that is part of the defined behavior).

Other validations should remain check-only initially, especially max line length and indentation, because automatically changing those can alter semantics. Converting leading tabs to spaces could likewise be a separate, explicitly opt-in fix once its behavior is well-defined.

Would maintainers welcome a focused PR that introduces a `--fix` mode for the safe rules first, with tests and clear documentation of which rules are/are not autofixable? I would be happy to work from the old `addFix` branch where useful, while keeping the initial scope deliberately conservative.

**Comment by mstruebing** [comment]
Yes we are definitely open to contributions in that regard. I personally also have the opinion that incomplete autofix is better than nothing.
